### PR TITLE
fix(start_planner): fix path generation for separated lanes and extended loop lanes

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

solve https://github.com/autowarefoundation/autoware.universe/issues/4391

image

The current_lanes variable currently only includes lanes within a forward_path_length of 300m, and next lanes are determined by extendNextLane. However, if there is more than one candidate for the next lane, and if an incorrect lane is added, the arc coordinates of the goal cannot be calculated correctly. In this PR, we've added the next lane in route_lanes to the extended lanes, if that lane exists. We've also fixed the loop check to only verify start lanes rather than all extended lanes. This is because sometimes the target next lane for extension is the same as the previously extended lanes, which causes a failure in adding the lane. This same issue can result in goal arc coordinates not being calculated correctly.

## Related links

solve https://github.com/autowarefoundation/autoware.universe/issues/4391

## Tests performed

[tier4 internal scenario test](https://evaluation.tier4.jp/evaluation/reports/bbc8f39a-8f83-59f3-a725-fa4771960e35?project_id=prd_jt) 1580/1588

base:
1577/1588([2023/07/30](https://evaluation.tier4.jp/evaluation/reports/d1ec5471-3332-5f7c-9203-6d40ef0f270e/?project_id=prd_jt))

psim

the issues separated next lanes case 

image

long overlapped route lanes case (extended previous lanes are same to candidate of extended next lanes)

before

image

after 

image

## Interface changes

none
## Effects on system behavior

none

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kosuke55** [comment]
this PR causes another loop path generation issues
need https://github.com/autowarefoundation/autoware.universe/pull/4422 to fix that

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM
It's better why loop check is necessary i thought

**Comment by kosuke55** [comment]
> It's better why loop check is necessary i thought

thanks, I added comments in 30a0f27bfaec174c802dcf12dea65af8315061a2

File: planning/behavior_path_planner/src/utils/utils.cpp [hunk 1]
```diff
@@ -2854,7 +2854,14 @@ lanelet::ConstLanelets extendNextLane(
   // Add next lane
   const auto next_lanes = route_handler->getNextLanelets(extended_lanes.back());
   if (!next_lanes.empty()) {
-    extended_lanes.push_back(next_lanes.front());
+    // use the next lane in route if it exists
```
[the remark this answers is not in the document]
Comment (kosuke55) [reply planning/behavior_path_planner/src/utils/utils.cpp:2857]: > long overlapped route lanes case (extended previous lanes are same to candidate of extended next lanes)
yes this and `loop check` with only start lane solved the problem

> I'm just curious why this happened thinking
like this case, if extended next candidate1 is selected, the lane is same to extended prev so loop_check is caught and extend is terminated.
For this case I changed the loop check not with all laens but only start lanes.

image

[the remark this answers is not in the document]
Comment (kosuke55) [reply planning/behavior_path_planner/src/utils/utils.cpp:2857]: this is a better example where loop_check is always caught and extend is terminated.

image

[the remark this answers is not in the document]
Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/src/utils/utils.cpp:2857]: Thank you for the very clear explanation.

**Pull request merged**